src/juju/reducers.js

```diff
@@ -53,7 +53,13 @@ export default produce(
         // There don't appear to be any irrelevent data in the modelInfo so
         // we overwrite the whole object every time it changes even though
         // mostly that'll just be status timestamps.
-        draftState.modelData[modelInfo.uuid].info = modelInfo;
+        const modelData = draftState.modelData[modelInfo.uuid];
+        // If any of the status requests timeout then it's possible the data
+        // won't be available. Just abandon saving any data in that case.
+        // This will go away with the new API.
+        if (modelData) {
+          draftState.modelData[modelInfo.uuid].info = modelInfo;
```

Comment:
Why not use `modelData.infro = modelInfo;` instead of accessing the full object again? It made me reread this conditional to make sense of it.

Reply:
I was being explicit because `draftState` is a proxy supplied by the library `immer` that we use for immutable state.